src/pudl/extract/ferc1.py

```diff
@@ -213,429 +208,272 @@
+
+    def get_db_path(self) -> str:
+        """Returns the connection string for the sqlite database."""
+        db_path = str(Path(self.output_path) / self.DATABASE_NAME)
+        return f"sqlite:///{db_path}"
+
+    def execute(self):
+        """Runs the extraction of the data from dbf to sqlite."""
+        # TODO(rousik): perhaps we should check clobber here before starting anything.
+        self.delete_schema()
+        self.create_sqlite_tables()
+        self.load_table_data()
+        self.postprocess()
+
+    def delete_schema(self):
+        """Drops all tables from the existing sqlite database."""
+        try:
+            pudl.helpers.drop_tables(
+                self.sqlite_engine,
+                clobber=self.clobber,
             )
-        else:
-            records.append(
-                {
-                    "respondent_id": rid,
-                    "respondent_name": f"Missing Respondent {rid}",
-                },
+        except sa.exc.OperationalError:
+            pass
+        self.sqlite_engine = sa.create_engine(self.get_db_path())
+        self.sqlite_meta = sa.MetaData()
+        self.sqlite_meta.reflect(self.sqlite_engine)
+
+    def create_sqlite_tables(self):
+        """Creates database schema based on the input tables."""
+        refyear = self.settings.refyear
+        if refyear is None:
+            refyear = max(
+                DataSource.from_id(self.datastore.get_dataset()).working_partitions[
+                    "years"
+                ]
             )
-    return records
-
-
-def observed_respondents(ferc1_engine: sa.engine.Engine) -> set[int]:
-    """Compile the set of all observed respondent IDs found in the FERC 1 database.
-
-    A significant number of FERC 1 respondent IDs appear in the data tables, but not
-    in the f1_respondent_id table. In order to construct a self-consistent database with
-    we need to find all of those missing respondent IDs and inject them into the table
-    when we clone the database.
-
-    Args:
-        ferc1_engine: An engine for connecting to the FERC 1 database.
-
-    Returns:
-        Every respondent ID reported in any of the FERC 1 DB tables.
-    """
-    f1_table_meta = pudl.output.pudltabl.get_table_meta(ferc1_engine)
-    observed = set()
-    for table in f1_table_meta.values():
-        if "respondent_id" in table.columns:
-            observed = observed.union(
-                set(
-                    pd.read_sql_table(
-                        table.name, ferc1_engine, columns=["respondent_id"]
-                    ).respondent_id
-                )
+        for tn in self.datastore.get_table_names():
+            self.datastore.get_table_schema(tn, refyear).to_sqlite_table(
+                self.sqlite_meta
             )
-    return observed
+        self.finalize_schema(self.sqlite_meta)
+        self.sqlite_meta.create_all(self.sqlite_engine)
 
+    def transform_table(self, table_name: str, in_df: pd.DataFrame) -> pd.DataFrame:
```

Comment:
As this is set up right now, would we need to have a big if/elsif/else block to do the right thing for each table, if the operations weren't uniform across all tables?

Reply:
Right now yes. I haven't really gotten to it yet, but I think this should be replaced with a map of table transformation `dict[str,Callable[[pd.DataFrame],pd.DataFrame]]`, where for each of these extractor, you can simply register all the transformers for each table, or possibly use reflections and find all methods with names such as `transform_foo` as a transformation for the table `foo`.

I was planning on refactoring this when it becomes more necessary/relevant (i.e. when there are more transformations needed)

Reply:
Okay that sounds totally reasonable. We're trying to minimize the changes that happen in this step anyway (and deal with all the real transformations when we go from the cloned FERC DB into the main PUDL DB) so hopefully there won't actually very many cases to manage in here. Maybe even just one, for the `respondent_id` table.